### ISSUE TITLE
Vue js totally lacks coverage of CDN Option

### ISSUE THREAD

**Issue by murugappanrm** [vuejs/docs#1953]
Hi @bencodezen , 

This issue is branch of the issue discussed under https://github.com/vuejs/docs/issues/786#issuecomment-1241960990 so that it can be handled separately. The documentation issue is detailed in clearer parts:

**(1) Why provide the CDN Option?** 

The documentation in CDN usage is so skimpy that it gives us a feeling that the creators do have the intention to promote this option. Why then is this option available. All we get is one topic "Without Build Tools " under Quick Start. 

**(2) Questions not answered**
Numerous questions have been raised but not addressed in the documentation. The following question need to be addressed:
- with new developments, how do we use composition API with CDN
- There is no inkling on how to create components and load them like we do under CLI, Definining the CDN and creating the Vue app in each HTML page defeats the purpose of using the CDN with vue features. 

In simple terms. the creators either fully document all the features which can used under CDN option as well or drop CDN option altogether and put it out of misery.

**Comment by yyx990803** [comment]
The section you mentioned - [Quick Start - Without Build Tools](https://vuejs.org/guide/quick-start.html#without-build-tools) - explains exactly how to use Vue from a CDN (unpkg.com). You can change the unpkg URL to any other CDN that supports npm-published packages (for example JSDelivr).

All [Examples](https://vuejs.org/examples/), with API preference set to "HTML", are also using Vue from a CDN.

All Vue APIs work the same between CDN and non-CDN usage, except SFC-specific features. You can't use SFCs when using Vue from a CDN because it's way too costly to ship the entire SFC compiler to the browser.

All non-SFC code samples throughout the entire documentation apply to CDN-based usage. As for your question regarding using Composition API in non-SFC context, by using the `setup()` option instead of `<script setup>`. This is covered as the first thing in [Reactivity Fundamentals](https://vuejs.org/guide/essentials/reactivity-fundamentals.html) (when API preference is Composition).

To sum up - everything you are asking for is in fact covered. Maybe we just didn't make it obvious enough. We will think about how to improve that.

That said, do realize you are using free, open source software. When interacting with maintainers, keep in mind you are not a paying customer, and GitHub issues are not customer service. Maintainers of the project do not owe you anything. GitHub support is provided as a courtesy, not an obligation. We all have many things on our plate and we can't guarantee answering every issue immediately. If you want to use other alternatives like Alpine or Astro because of this, feel free to do so.

**Comment by bencodezen** [comment]
Hi @murugappanrm. As Evan has mentioned in his response, while we always look for ways to improve on what is documented, it's important to understand that the tone of the way this issue was filed is problematic because it comes off in a way that can be perceived as malicious towards the maintainers (many of whom who do this on a volunteer basis and simply want to do their best for the community).

In the future, when filing issues, I'd advise against statements that talk in absolute terms given that things rarely are 100% any way. For example, the issue title claims that "CDN Option is missing entirely" which is not true. In addition, statements like "Creators should fully document features or put it out of its misery" also is unnecessarily hostile and fails to acknowledge the reality that "perfection is the enemy of progress." We always aim to document things as best as we can, but we are human and would never claim to know every use case in how people would interpret docs. So iteration is forever a part of our docs process. 

Finally, while I understand there are frustrations with things being confusing in your journey, the community of open source is really only sustainable if built on principles of collaboration and kindness. For an example of how this might be done, see the new issue https://github.com/vuejs/docs/issues/1957 here on how an issue can be raised without attacking the maintainers or judging decisions without the full context of why decisions were made.

**Comment by murugappanrm** [comment]
@bencodezen @yyx990803 ,

Please do tell me which part of post is "malicious"? I am just "calling a spade a spade" as I, a user, see it. Sometimes we may need to be less emotional and more objective. Again this is just my view. Nowhere in the my post i was unappreciative. I raised the issue as i and one other person in previous post see lacking in the documentation. 

Please do realize, writing a documentation in Einstein formula makes absolutely no sense to newbie who is just learning mathematics. I am newbie and reading a documentation and expecting me to understand things at the level of a super-hero would be unfair, would you agree? I am a "zero" who want to be a "hero".  Please go read the documentation from a newbie perspective and advise If i am wrong.  If i am wrong, then, as @yyx990803 says, we should looks elsewhere as the  "free-software" may only be for use of "scientists". Which i am not cause i have realistic and have my 2 feet on the ground. I learnt one thing, irrespective of paid or unpaid, the success of a software is based on the pervasiveness and ease of adoption amongst newbies. May be i am wrong or may hurt the emotionally, i am sorry.

@bencodezen , "totally lacks coverage" and  "CDN Option is missing entirely" are not the same thing. If someone says Vuejs is a "crap" in the YouTube, would you run him down into the earth? Its his way of expressing the  frustration after being told that everything is well covered in the documentation and then they go loss of words when requested for clarity. Millions of time i have called my own documentation "$hit" when my developers dont understand the crap written in it. Does that mean they are fools or I am the idiot? So lets be more objective and less emotional.  Me saying "drop CDN option altogether and put it out of misery" is as a result of many telling us to stop using CDN as use CLI without any absolute clue why we need to use the CDN, implicitly saying "CDN way is incorrect".

Anyway thank you for taking the trouble to explain that just because it is open-source i should not express myself freely.

**Comment by pal-software** [comment]
Hmm...
Whatever you say, [murugappanrm](https://github.com/murugappanrm) was right. The lack of sufficient documentation (or examples) simply does not allow sometimes to make the transition to a new version (at least for now).
In my current prod app I am using SFC (webpack) & vue 2 with vue-router.
But when I tried migrating to vue 3 it became a problem.
Our current application works on the one hand as a regular web application (http/https protocols), and inside a native wrapper application (Android/Apple), where it opens using the "file:///" protocol (so we can't use newfangled vite).
When I tried to configure the build via webpack, I found problems with the output of the information "The resolve component can only be used in render() or setup().". I can resolve this with my components, but I can't resolve this for vue-router (with router0view, router-lonk & etc).
So, now I have to stay on vue2 simply because I can't spend a huge amount of time searching for a solution..
It's just a statement of fact.

**Comment by pal-software** [comment]
> Finally, while I understand there are frustrations with things being confusing in your journey, the community of open source is really only sustainable if built on principles of collaboration and kindness. For an example of how this might be done, see the new issue #1957 here on how an issue can be raised without attacking the maintainers or judging decisions without the full context of why decisions were made.

Unfortunately, you gave a bad example. The question is asked, but the real answer (result) is not received.

**Comment by murugappanrm** [comment]
@pal-software 

I gave up on this subject because I got nowhere with it as it came down to  emotions and pride rather than reality and truth. I understand that this open source and its free but they do not realize that this piece happens to the base for many frameworks such as Quasar, Vuetify, Nuxt etc. As such it all the more of utmost importance that the documentation should be comprehensive and complete. Open source does not mean immunity to criticism.

Anyway, its a lost cause. Many of have decided to move on to vanilla javascript, Astro. Alpine.js, React and the god-given php (which is real programming for the server side). We realized this piece is just a re-invention of the wheel. 

Good luck @pal-software .

**Comment by bencodezen** [comment]
@pal-software I absolutely agree that we can do better to improve the documentation to cover ways for people to migrate and use different methodologies. I am not satisfied and certainly hope to make some progress on this.

If possible, could you file a new issue with where you're having complications with the migration? Would love to try and brainstorm ways we can improve on this for future users.

**Comment by pal-software** [comment]
> @pal-software I absolutely agree that we can do better to improve the documentation to cover ways for people to migrate and use different methodologies. I am not satisfied and certainly hope to make some progress on this.
> 
> If possible, could you file a new issue with where you're having complications with the migration? Would love to try and brainstorm ways we can improve on this for future users.

#2394
